From 9708cca9fac2b9fd13803118ddc7af6b28549e1a Mon Sep 17 00:00:00 2001
From: Pascal Nowack <Pascal.Nowack@gmx.de>
Date: Sun, 14 Apr 2024 17:15:30 +0200
Subject: [PATCH] rdp/rdp-plugin: Fix faulty color depth check

The color depth setting is a uint32 setting and thus requires the
respective getter. Also, it expects a settings key, not some random
boolean condition.
Thankfully, FreeRDP3 introduced some assertions around the setting
types, making this issue discoverable.

Closes: https://gitlab.com/Remmina/Remmina/-/issues/3090
---
 plugins/rdp/rdp_plugin.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/plugins/rdp/rdp_plugin.c b/plugins/rdp/rdp_plugin.c
index 1e691ea58..2513b4b58 100644
--- a/plugins/rdp/rdp_plugin.c
+++ b/plugins/rdp/rdp_plugin.c
@@ -1658,7 +1658,9 @@ static gboolean remmina_rdp_main(RemminaProtocolWidget *gp)
 	REMMINA_PLUGIN_DEBUG("gfx_h264_available: %d", gfx_h264_available);
 
 	/* Avoid using H.264 modes if they are not available on libfreerdp */
-	if (!gfx_h264_available && (freerdp_settings_get_uint32(rfi->clientContext.context.settings, FreeRDP_ColorDepth) == 65 || freerdp_settings_get_bool(rfi->clientContext.context.settings, FreeRDP_ColorDepth == 66)))
+	if (!gfx_h264_available &&
+	    (freerdp_settings_get_uint32(rfi->clientContext.context.settings, FreeRDP_ColorDepth) == 65 ||
+	     freerdp_settings_get_uint32(rfi->clientContext.context.settings, FreeRDP_ColorDepth) == 66))
 		freerdp_settings_set_uint32(rfi->clientContext.context.settings, FreeRDP_ColorDepth, 64); // Fallback to GFX RFX
 
 	if (freerdp_settings_get_uint32(rfi->clientContext.context.settings, FreeRDP_ColorDepth) == 0) {
-- 
GitLab

